packages/harpguru-core/src/components/new-harpstrata-state-callback/hooks/use-derive-columnbounds/use-derive-columnbounds.ts

```diff
@@ -0,0 +1,15 @@
+import type { HarpStrata } from 'harpstrata'
+
+import { deriveColumnBounds } from '../../utils'
+import { compareColumnBounds } from '../../../../utils'
+
+export const useDeriveColumnBounds = (
+  newHarpStrata: HarpStrata,
+  prevColumnBounds: 'FIT' | readonly [number, number],
+  setColumnBounds: (arg0: 'FIT' | readonly [number, number]) => void
+): 'FIT' | readonly [number, number] => {
+  const nextColumnBounds = deriveColumnBounds(newHarpStrata, prevColumnBounds)
+  if (!compareColumnBounds(prevColumnBounds, nextColumnBounds))
+    setColumnBounds(nextColumnBounds)
+  return nextColumnBounds
+}
```

Comment:
As well as being renamed and moved to utils folder, we should make tests for all of them.

Reply:
Raised as techdebt in https://github.com/js-jslog/harpguru/issues/120#issuecomment-938452028